operator/pkg/ciliumendpointbatch/manager.go

```diff
@@ -512,9 +521,15 @@ func (c *cebManagerIdentity) InsertCepInCache(cep *cilium_v2.CoreCiliumEndpoint)
 	// check the given cep is already exists in any of the CEB.
 	// if yes, Update a ceb with the given cep object.
 	if cebName, exists := c.cacheCepMap.get(GetCepNameFromCCEP(cep)); exists {
-		// add a cep into the ceb
-		c.addCEPtoCEB(cep, cebName)
-		return cebName
+		// If CEP identity is changed, remove the old CEP from cache and insert
+		// the CEP in a new CEB
+		if c.cebToIdentity[cebName] != cep.IdentityID {
+			c.RemoveCepFromCache(GetCepNameFromCCEP(cep))
```

Comment:
Don't we also need to remove the CEP from the actual CEB (not just from the cache)?

Reply:
Updated the comment. `RemoveCepFromCache` will remove the CEP from local CEB cache and will trigger an UPDATE to k8s-apiserver to sync local CEB with k8s-apiserver.